### PR TITLE
feat: Add option for double-matching in CKF performance writer

### PR DESCRIPTION
**Pull request by benjaminhuth**
So far we do single-matching in the `CkfPerformanceWriter`, i.e., we associate to matched and fake based on the track purity of the reconstructed track. This is not always reliable.
This adds an option for double matching, which checks also the efficiency of the truth track.

### CODE REVIEW

File: Examples/Io/Performance/ActsExamples/Io/Performance/CKFPerformanceWriter.hpp [hunk 1]
```diff
@@ -66,6 +66,9 @@ class CKFPerformanceWriter final : public WriterT<TrajectoriesContainer> {
     DuplicationPlotTool::Config duplicationPlotToolConfig;
     TrackSummaryPlotTool::Config trackSummaryPlotToolConfig;
 
+    /// Whether to do double matching or not
+    bool doubleMatching = false;
```
Comment (andiwand) [inline Examples/Io/Performance/ActsExamples/Io/Performance/CKFPerformanceWriter.hpp:70]: does it make sense to make this default on?

**Review by andiwand — APPROVED** [review]

File: Examples/Io/Performance/ActsExamples/Io/Performance/CKFPerformanceWriter.hpp (hunk 1 above)
Comment (benjaminhuth) [reply Examples/Io/Performance/ActsExamples/Io/Performance/CKFPerformanceWriter.hpp:70]: I didn't want to break any hashes... do we have this in our physmon?
In principle I would make this the default...

Comment (andiwand) [reply Examples/Io/Performance/ActsExamples/Io/Performance/CKFPerformanceWriter.hpp:70]: yes I think this would change a bunch of reference files

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6358939)!

|status|job|
|------|---|
|:red_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281812) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281811) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281810) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281809) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281808) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281807) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281806) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281805) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281804) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281803) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281802) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281801) |
|:red_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281800) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281799) |
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281798) |
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281797) |
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33281796) |

**Pull request merged**